pkg/api/types.go

```diff
@@ -476,6 +478,18 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// MetadataVolumeSource represents an host directory mounted on a POD. It stores pod's metadata info
+type MetadataVolumeSource struct {
+	// PodNameFileName is the name of the file where name of the pod is written
+	PodNameFileName string `json:"podNameFileName,omitempty"`
```

Comment:
@thockin @sdminonne I have been thinking that this and #6379 should use the same API.  We could have the volume API be something like:

```go
type MetadataVolumeSource struct {
  Files []MetadataFile
}

type MetadataFile struct {
  Source ObjectReference
  Path   string
  Mode   os.FileMode
}
```

Where `ObjectInfo` is analogous with the way we currently handle addressing the field of the pod in #6379.

This is what I had been thinking when I mentioned having the env and volume downward APIs use the same mechanisms on the last community hangout.